addons/website/models/website.py

```diff
@@ -1837,6 +1842,14 @@ def _trigram_enumerate_words(self, search_details, search, limit):
                         value = value.lower()
                         yield from re.findall(match_pattern, value)
 
+    # As _trigram_enumerate_words use a hardcoded sql query, it needs some filter regarding performance. (Especially for product_template)
+    def _trigram_enumerate_words_where_clause(self, model_name):
+        return "WHERE True"
+
+    # As _trigram_enumerate_words use a hardcoded sql query, it needs some filter regarding performance. (Especially for product_template)
+    def _trigram_enumerate_words_published_filter(self, model_name):
```

Comment:
Could also be named something like `_where_clause`, so you encapsulate every part of the WHERE condition in this method, and it will not be just about the publish condition, should anyone want to add its own unrelated condition, it will be a generic method to put the condition into.